Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -298,15 +300,18 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            guard let connection = self["connection"].first?.lowercased() else {
-                // HTTP 1.1 use keep-alive by default if not otherwise told.
-                return version.major == 1 && version.minor == 1
-            }
-            
-            if connection == "close" {
-                return false
+            if !self.headers.isEmpty {
+                for header in self.headers {
+                    if self.buffer.equalCaseInsensitiveASCII(view: connectionUtf8, at: header.name) {
+                        if self.buffer.equalCaseInsensitiveASCII(view: closeUtf8, at: header.value) {
+                            return false
+                        }
+                        return self.buffer.equalCaseInsensitiveASCII(view: keepAliveUtf8, at: header.value)
+                    }
+                }
             }
-            return connection == "keep-alive"
+            // HTTP 1.1 use keep-alive by default if not otherwise told.
+            return version.major == 1 && version.minor == 1
```

Comment:
this should be `version.minor >= 1` as RFC says to treat 1.x where x>=1 like 1.1